### ISSUE TITLE
[RRFC] ability to run scripts standalone, without their pre- and post- associates

### ISSUE THREAD

**Issue by morgaan** [npm/rfcs#158]
## Motivation ("The Why")

To provide a way to run npm script without the pre- and/or post- scripts

### When

When running `npm run <script>`

### Where

npm cli

### How

By passing flags for example:

- `npm run test --no-pre`, or
- `npm run test --no-post`, or
- `npm run test --no-pre-post`

#### Current Behaviour

pre- and post- scripts cannot be skipped, workaround exist but clumsy, for example:

```
"scripts": {
    "rawtest": "mocha -r ts-node/register src/**/*.spec.ts",
    "pretest": "<some flow that isn't always desired>"
    "test": "npm run rawtest",
  }
```

#### Desired Behaviour

- the pre- scripts should be skipped when run `npm run <script> --no-pre`
- the post- scripts should be skipped when run `npm run <script> --no-post`
- both pre- and post- scripts should be skipped when run `npm run <script> --no-pre-post`

### References

Was added by @beenotung there https://github.com/npm/cli/issues/484

**Comment by gingur** [comment]
I think this could also solve https://npm.community/t/ignore-scripts-ignores-all-scripts/1403 , being able to run `npm install` without "post" scripts (but let dependency installs work as expected)

**Comment by askirmas** [comment]
I found 2 better syntaxes until it is not implemented on core level
```json
{
  "test--dry": "mocha",
  "start --dry": "node server"
}
```
```bash
npm run test--dry && npm run start\ --dry
```
Besides, [git run without hooks](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt--n)

**Comment by ljharb** [comment]
@askirmas `npm run start -- --raw` and `npm test -- --raw` works without all those `--` hijinks in the scripts.

**Comment by askirmas** [comment]
> @askirmas `npm run start -- --raw` and `npm test -- --raw` works without all those `--` hijinks in the scripts.

`start --raw` will run everything: `prestart`, `start` with arg `--raw` and `poststart`. That's not the point of issue

**Comment by ljharb** [comment]
In that case, I use your technique often :-) about 240 of my projects all use `npm run tests-only` for just the tests, because `npm test` invokes `npm run pretest` and `npm run posttest`, and in CI, i don't want the linter running on the 100 node versions i want the tests running, for example.

**Comment by askirmas** [comment]
But this hack makes *package.json* to look dirty.

**Comment by ljharb** [comment]
I don't think a single extra script looks "dirty", especially not compared to a bunch of extra double-dashes everywhere - but obviously the cleanest would be a way to do `npm test --ignore-pre-post-scripts` or similar.

**Comment by askirmas** [comment]
Here I guess

https://github.com/npm/cli/blob/656bce7dd0f9753a273912e803261ed246593924/lib/run-script.js#L79

**Comment by beenotung** [comment]
I think we can (easily?) write a script to read from the package.json and execute the command from local node_module or PATH in env according to archive what we want, once we made it, npm may adopt our impl into the core

**Comment by ruyadorno** [comment]
We have implemented #185 in the the v7 beta which does what is described here but with a different syntax:

```
npm run <script-name> --ignore-scripts
```
